### PR TITLE
Add LL/CON/ADV/BV-14-C and LL/CON/ADV/BV-15-C test cases

### PR DESCRIPTION
**Pull request by Tronil**
Added the two test cases

Updated PacketDecoder to support a lot more packets and added a flush() to clear out any existing packets when looping inside a test case

Updated Initiator to support connecting via extended commands

Signed-off-by: Troels Nilsson <trnn@demant.com>

### CODE REVIEW

File: src/components/dump.py [hunk 1]
```diff
@@ -193,22 +205,123 @@ def connect_ind(payload):
...
+            payload_type = PacketType.ADV_UNKNOWN_PDU
+            payload = data
     else:
-        return None
+        if header.PDU_Type == 0b0011:
+            payload_type = PacketType.AUX_SCAN_REQ
+            Payload = namedtuple('Payload', 'AdvA, TargetA')
+            advA = int.from_bytes(data[:6], 'little', signed=False)
+            targetA = int.from_bytes(data[6:], 'little', signed=False)
+            payload = Payload(advA, targetA)
```
[the remark this answers is not in the document]
Comment (Tronil) [reply src/components/dump.py:280]: Yes - the AUX_SCAN_REQ does not use the Common Extended Advertising Payload; It is the same payload as a legacy SCAN_REQ

File: src/components/dump.py [hunk 2]
```diff
@@ -193,22 +205,123 @@ def connect_ind(payload):
...
+    return PacketType.ADV_EXT_UNKNOWN_PDU
+
+def aux_ptr_matches(aux_ptr, superior_packet_ts, ts, channel_num, phy):
+    ch = channel_num_to_index(channel_num)
+    if ch == aux_ptr.chIdx and ((phy == '1M' and (aux_ptr.auxPHY == 0x00 or aux_ptr.auxPHY == 0x02)) or (phy == '2M' and aux_ptr.auxPHY == 0x01)):
+        if (aux_ptr.auxOffset > 0): # AuxOffset == 0 means no auxillary packet will be transmitted
+            t_start_offset = (300 if aux_ptr.offsetUnits == 1 else 30) * aux_ptr.auxOffset
+            t_end_offset = (300 if aux_ptr.offsetUnits == 1 else 30) * (aux_ptr.auxOffset + 1)
+            # Adjust according to clock accuracy value
+            ca_adjustment = math.ceil(t_start_offset * ((50 if aux_ptr.CA == 1 else 500)/1000000))
```
[the remark this answers is not in the document]
Comment (Tronil) [reply src/components/dump.py:319]: True - I'll fix that up

File: src/tests/ll_verification.py [hunk 3]
```diff
@@ -1540,6 +1541,197 @@ def ll_con_adv_bv_10_c(transport, upperTester, lowerTester, trace):
 
     return success;
 
+# Implements LL/CON/ADV/BV-14-C and LL/CON/ADV/BV-15-C test cases (only difference is the PHY)
+def do_ll_con_adv_bv_14_15_16_c(transport, upperTester, lowerTester, trace, packets, phy):
```
[the remark this answers is not in the document]
Comment (Tronil) [reply src/tests/ll_verification.py:1545]: Yes - 16 is for the coded PHY, but I later realized some of the calculations don't work as-is for coded PHY. I'll rename the function to reflect that

**Comment by wopu-ot** [comment]
LGTM

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**